native/ortex/src/model.rs

```diff
@@ -83,27 +84,46 @@ pub fn run(
     model: ResourceArc<OrtexModel>,
     inputs: Vec<ResourceArc<OrtexTensor>>,
 ) -> Result<Vec<(ResourceArc<OrtexTensor>, Vec<usize>, Atom, usize)>, Error> {
-    // TODO: can we handle an error more elegantly than just .unwrap()?
+    // Grab the session and run a forward pass with it
+    let session: &ort::Session = &model.session;
 
     let mut ortified_inputs: Vec<ort::SessionInputValue> = Vec::new();
-    for input in inputs {
-        let derefed_input: &OrtexTensor = &input;
-        let v: ort::SessionInputValue = derefed_input.try_into()?;
-        ortified_inputs.push(v);
-    }
 
-    // Grab the session and run a forward pass with it
-    let session: &ort::Session = &model.session;
+    for (elixir_input, onnx_input) in zip(inputs, &session.inputs) {
+        let derefed_input: &OrtexTensor = &elixir_input;
+        if is_bool_input(&onnx_input.input_type) {
+            // this assumes that the boolean input isn't huge -- we're cloning it twice;
+            // once below, once in the try_into()
+            let boolified_input: &OrtexTensor = &derefed_input.clone().to_bool();
+            let v: ort::SessionInputValue = boolified_input.try_into()?;
+            ortified_inputs.push(v);
+        } else {
+            let v: ort::SessionInputValue = derefed_input.try_into()?;
+            ortified_inputs.push(v);
+        }
+    }
 
     // Construct a Vec of ModelOutput enums based on the DynOrtTensor data type
     let outputs = session.run(&ortified_inputs[..])?;
-    outputs
-        .iter()
-        .map(|(_name, val)| {
-            let ortextensor: OrtexTensor = OrtexTensor::try_from(val)?;
-            let shape = ortextensor.shape();
-            let (dtype, bits) = ortextensor.dtype();
-            Ok((ResourceArc::new(ortextensor), shape, dtype, bits))
-        })
-        .collect()
+    let mut collected_outputs = Vec::new();
+
+    for output_descriptor in &session.outputs {
+        let output_name: &str = &output_descriptor.name;
+        let val = outputs.get(output_name).expect(
+            &format!(
+                "Expected {} to be in the outputs, but didn't find it",
+                output_name
+            )[..],
+        );
+
+        // NOTE: try_into impl here will implicitly map bool outputs to signed i8 outputs
```

Comment:
If we map to a `u8` here (see other comment), it will be in line with how Nx treats booleans.